cmd/thanos/rule.go

```diff
@@ -139,15 +147,42 @@ func runRule(
 		return nil, errors.Errorf("no query peer reachable")
 	}
 
-	var mgr *rules.Manager
-
+	// Run rule evaluation and alert notifications.
+	var (
+		alertmgrs = newAlertmanagerSet(alertmgrURLs, nil)
+		alertQ    = alert.NewQueue(logger, reg, 10000, 100, labelsTSDBToProm(lset))
+		mgr       *rules.Manager
+	)
 	{
 		ctx, cancel := context.WithCancel(context.Background())
 
+		notify := func(ctx context.Context, expr string, alerts ...*rules.Alert) error {
+			res := make([]*alert.Alert, 0, len(alerts))
+			for _, alrt := range alerts {
+				// Only send actually firing alerts.
+				if alrt.State == rules.StatePending {
```

Comment:
`alrt.State != rules.StateFiring` instead? There is some `StateInactive` also

Reply:
This is the logic as found in upstream. We shouldn't diverge from it.
But you are right, it would make more sense and _should_ be equivalent. I can bring it up upstream.